pkg/config/config.go

```diff
@@ -555,21 +555,35 @@ func (c *GlobalConfigInfo) DeleteApplication(application string, project string)
 		c.ActiveApplications = []ApplicationInfo{}
 	}
 
-	found := false
+	foundAt := -1
+	nextFoundAt := -1
+	isDeletedAppActive := false
 	for i, app := range c.ActiveApplications {
-		// if application exists set is as Active
+		// if application exists, save the index for deletion later and check if it is the active application
 		if app.Name == application && app.Project == project {
-			// remove current item from array
-			c.ActiveApplications = append(c.ActiveApplications[:i], c.ActiveApplications[i+1:]...)
-			found = true
+			isDeletedAppActive = app.Active
+			foundAt = i
+		}
+
+		// find the first other app in the same project
+		if app.Name != application && app.Project == project && nextFoundAt == -1 {
+			nextFoundAt = i
 		}
 	}
 
-	if !found {
+	if foundAt == -1 {
 		return fmt.Errorf("application %s doesn't exist", application)
 
 	}
 
+	// if the deleted app is the active application then set the next app in the project as active
+	if nextFoundAt != -1 && isDeletedAppActive {
```

Comment:
+1 this should ensure that if application being deleted is the last application, then no active application is set